src/diag/map_editor.cpp

```diff
@@ -5,149 +5,96 @@
-            return MAP_NAME_S_PETROL_UPSHIFT;
-        case MAP_ID_S_PETROL_DOWNSHIFT:
-            return MAP_NAME_S_PETROL_DOWNSHIFT;
-        case MAP_ID_C_DIESEL_UPSHIFT:
-            return MAP_NAME_C_DIESEL_UPSHIFT;
-        case MAP_ID_C_DIESEL_DOWNSHIFT:
-            return MAP_NAME_C_DIESEL_DOWNSHIFT;
-        case MAP_ID_C_PETROL_UPSHIFT:
-            return MAP_NAME_C_PETROL_UPSHIFT;
-        case MAP_ID_C_PETROL_DOWNSHIFT:
-            return MAP_NAME_C_PETROL_DOWNSHIFT;
-        case MAP_ID_A_DIESEL_UPSHIFT:
-            return MAP_NAME_A_DIESEL_UPSHIFT;
-        case MAP_ID_A_DIESEL_DOWNSHIFT:
-            return MAP_NAME_A_DIESEL_DOWNSHIFT;
-        case MAP_ID_A_PETROL_UPSHIFT:
-            return MAP_NAME_A_PETROL_UPSHIFT;
-        case MAP_ID_A_PETROL_DOWNSHIFT:
-            return MAP_NAME_A_PETROL_DOWNSHIFT; 
+        case A_UPSHIFT_MAP_ID:
+            return agility->get_upshift_map();
+        case C_UPSHIFT_MAP_ID:
+            return comfort->get_upshift_map();
+        case S_UPSHIFT_MAP_ID:
+            return standard->get_upshift_map();
+        case A_DOWNSHIFT_MAP_ID:
+            return agility->get_downshift_map();
+        case C_DOWNSHIFT_MAP_ID:
+            return comfort->get_downshift_map();
+        case S_DOWNSHIFT_MAP_ID:
+            return standard->get_downshift_map();
+        case WORKING_PRESSURE_MAP_ID:
+            return pressure_manager->get_working_map();
+        case PCS_CURRENT_MAP_ID:
+            return pressure_manager->get_pcs_map();
+        case TCC_PWM_MAP_ID:
+            return pressure_manager->get_tcc_pwm_map();
+        case FILL_TIME_MAP_ID:
+            return pressure_manager->get_fill_time_map();
+        case FILL_PRESSURE_MAP_ID:
+            return pressure_manager->get_fill_time_map();
         default:
             return nullptr;
     }
 }
 
-const int16_t* get_default_map(uint8_t map_id) {
-    switch(map_id) {
-        case MAP_ID_S_DIESEL_UPSHIFT:
-            return S_DIESEL_UPSHIFT_MAP;
-        case MAP_ID_S_DIESEL_DOWNSHIFT:
-            return S_DIESEL_DOWNSHIFT_MAP;
-        case MAP_ID_S_PETROL_UPSHIFT:
-            return S_PETROL_UPSHIFT_MAP;
-        case MAP_ID_S_PETROL_DOWNSHIFT:
-            return S_PETROL_DOWNSHIFT_MAP;
-        case MAP_ID_C_DIESEL_UPSHIFT:
-            return C_DIESEL_UPSHIFT_MAP;
-        case MAP_ID_C_DIESEL_DOWNSHIFT:
-            return C_DIESEL_DOWNSHIFT_MAP;
-        case MAP_ID_C_PETROL_UPSHIFT:
-            return C_PETROL_UPSHIFT_MAP;
-        case MAP_ID_C_PETROL_DOWNSHIFT:
-            return C_PETROL_DOWNSHIFT_MAP;
-        case MAP_ID_A_DIESEL_UPSHIFT:
-            return A_DIESEL_UPSHIFT_MAP;
-        case MAP_ID_A_DIESEL_DOWNSHIFT:
-            return A_DIESEL_DOWNSHIFT_MAP;
-        case MAP_ID_A_PETROL_UPSHIFT:
-            return A_PETROL_UPSHIFT_MAP;
-        case MAP_ID_A_PETROL_DOWNSHIFT:
-            return A_PETROL_DOWNSHIFT_MAP;
-        default:
-            return nullptr;
+#define CHECK_MAP(map_id) \
```

Comment:
Consider rather to use an inline function instead of a complicated #define. A return within a #define is rather hard when debugging.